src/builder.rs

```diff
@@ -159,84 +141,52 @@ impl SessionBuilder {
-        }
-
-        if let Some(ref interval) = self.server_alive_interval {
-            init.arg("-o")
-                .arg(format!("ServerAliveInterval={}", interval));
-        }
-
-        if let Some(ref port) = self.port {
-            init.arg("-p").arg(port);
-        }
-
-        if let Some(ref user) = self.user {
-            init.arg("-l").arg(user);
-        }
-
-        if let Some(ref k) = self.keyfile {
-            // if the user gives a keyfile, _only_ use that keyfile
-            init.arg("-o").arg("IdentitiesOnly=yes");
-            init.arg("-i").arg(k);
-        }
-
-        if let Some(ref config_file) = self.config_file {
-            init.arg("-F").arg(config_file);
-        }
-
-        init.arg(destination);
-
-        // eprintln!("{:?}", init);
+    /// Connect to the host at the given `host` over SSH using process_impl, which will
+    /// spawn a new ssh process for each `Child` created.
+    ///
+    /// The format of `destination` is the same as the `destination` argument to `ssh`. It may be
+    /// specified as either `[user@]hostname` or a URI of the form `ssh://[user@]hostname[:port]`.
+    /// A username or port that is specified in the connection string overrides the one set in the
+    /// builder (but does not change the builder).
+    ///
+    /// If connecting requires interactive authentication based on `STDIN` (such as reading a
+    /// password), the connection will fail. Consider setting up keypair-based authentication
+    /// instead.
+    pub async fn connect<S: AsRef<str>>(&self, destination: S) -> Result<Session, Error> {
+        let destination = destination.as_ref();
+        let (builder, destination) = self.resolve(destination);
+        builder.just_connect(destination).await
+    }
 
-        // we spawn and immediately wait, because the process is supposed to fork.
-        // note that we cannot use .output, since it _also_ tries to read all of stdout/stderr.
-        // if the call _didn't_ error, then the backgrounded ssh client will still hold onto those
-        // handles, and it's still running, so those reads will hang indefinitely.
-        let mut child = init.spawn().map_err(Error::Connect)?;
-        let stdout = child.stdout.take().unwrap();
-        let mut stderr = child.stderr.take().unwrap();
-        let status = child.wait().await.map_err(Error::Connect)?;
+    pub(crate) async fn just_connect<S: AsRef<str>>(&self, host: S) -> Result<Session, Error> {
+        Ok(Session::new_process_imp(
+            super::process_impl::builder::just_connect(self, host).await?,
+        ))
+    }
 
-        if !status.success() {
-            let mut err = String::new();
-            stderr.read_to_string(&mut err).await.unwrap();
-            return Err(Error::interpret_ssh_error(&err));
-        }
+    /// Connect to the host at the given `host` over SSH using mux_client_impl, which will
+    /// create a new socket connection for each `Child` created.
```

Comment:
> And honestly, I feel like down the road we're probably going to scrap process_impl entirely and go just with mux, since it handles errors better. I just don't want to jump all the way in right now before we've had a chance to test it more thoroughly.

Agreed.

Reply:
> I think we should also tell users to read the crate-level documentation for details here, since this brief text is unlikely to fully capture the difference for them.

Documentation updated.